### ISSUE TITLE
Calamares Initialization Failed

### ISSUE THREAD

**Issue by a2ndTiberianBat** [JPyke3/mbp-manjaro#16]
**Mac Model**
MBP 16,1 (2019)

**Describe the bug**
Cannot install Manjaro due to Calamares Initialization Failure.

**To Reproduce**
1. Boot into USB after flashing with iso (I used the 5.7 pre-release version (KDE) under 'download a latest release' on the installation wiki).

2. Execute Step 7 on the installation wiki to sync time and install calamares-mbp.
-- I had to use -Sy instead of -S when installing calamares-mbp in order to install it with pacman

3. Open the installer and see Calamares Initialization Failed Screen (could not load modules partition@partition)

If steps 2 and 3 are reversed (opening installer and selecting keyboard, then installing calamares-mbp) it fails during the install (after doing the partitioning) and relaunching the installer gives the same error screen.

**Notes**
I used bootcamp to make a windows partition, then shrunk it to open enough unallocated space for manjaro (128gb) before starting the installation process.
I used the refind boot manager as I could not see the flash drive on the one that pops up after holding the options key while booting.

**Screenshots**
![calamares_error](https://user-images.githubusercontent.com/56736018/104131872-5a6fec80-5347-11eb-9951-aa8a27c33276.png)

**Comment by JPyke3** [comment]
Hi There!

Thank you for your issue report! The one thing I just want to double check before we go into a few more specifics, is are you running this command?:

```
sudo pacman -R calamares
```

before attempting to install `calamares-mbp`?

Secondly, I just wanted to double check that you are running `calamares` either by opening the desktop or menu application, or if you are launching it via a terminal that you are launching it as the root user.

**Comment by a2ndTiberianBat** [comment]
Hi! 

Yes, I am running that command before installing calamares-mbp.  I was opening the desktop 'install manjaro linux' application, should I try launching it from the terminal?

Thanks!

**Comment by JPyke3** [comment]
Yep I would give it a go!! Make sure you launch it as root:
```
sudo calamares 
```

If it doesn't work correctly. Put all the terminal output here as well :)

**Comment by a2ndTiberianBat** [comment]
![sudocalamares](https://user-images.githubusercontent.com/56736018/104132525-bf423b80-5375-11eb-9824-95e48f74242f.png)

**Comment by JPyke3** [comment]
Hm okay. We had this issue once before from someone else on the discord group, but he never provided an explanation as to how he fixed the issue.

I will grab a LiveUSB and attempt to replicate this, but I doubt I can do this until tomorrow. 

I suggest giving your whole machine a restart and attempting everything again from scratch, just as a last attempt until I can jump on and try tomorrow 👍

**Comment by a2ndTiberianBat** [comment]
Sounds good, thanks for the help!

**Comment by guest-08** [comment]
Hey,
I've have the same issue on a MBP 13'' (2018). 

Since yesterday, I've tried countless times, doing serveral differents things, none of them worked.

**1. From Git Clone**
I restarted from scratch and ran:
```
systemctl start systemd-timesyncd.service
sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
sudo pacman -R calamares
git clone https://github.com/JPyke3/calamares-mbp
cd calamares-mbp
makepkg -s
```
The compiling stopped saying I needed kconfig amongs other things.

**2. From git with kconfig**
I restarted the process and ran:
```
systemctl start systemd-timesyncd.service
sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
sudo pacman -R calamares
sudo pacman -Sy kconfig
git clone https://github.com/JPyke3/calamares-mbp
cd calamares-mbp
makepkg -s
```
This time, the process went futher but stopped with the following message:

> CMake Error at /usr/lib64/cmake/Qt5Qml/Qt5QmlConfig.cmake:112 (find_package):
>   Could not find a configuration file for package "Qt5Network" that is
>   compatible with requested version "5.15.2".
> 
>   The following configuration files were considered but not accepted:
> 
>     /usr/lib64/cmake/Qt5Network/Qt5NetworkConfig.cmake, version: 5.15.0
> 
> Call Stack (most recent call first):
>   /usr/lib64/cmake/Qt5QmlModels/Qt5QmlModelsConfig.cmake:112 (find_package)
>   /usr/lib64/cmake/Qt5Quick/Qt5QuickConfig.cmake:112 (find_package)
>   /usr/lib64/cmake/Qt5/Qt5Config.cmake:28 (find_package)
>   CMakeLists.txt:259 (find_package)
> 
> 
> -- Configuring incomplete, errors occurred!
> See also "/home/manjaro/calamares-mbp/src/calamares-3.2.13/build/CMakeFiles/CMakeOutput.log".
> ==> ERROR: A failure occurred in build().
>     Aborting...

**3. Cmake madness**
This is already way beyond my noob knowlegde, but still tried to troubleshoot. Since I got this message:

> CMake Error at /usr/lib64/cmake/Qt5Qml/Qt5QmlConfig.cmake:112 (find_package):
>   Could not find a configuration file for package "Qt5Network" that is
>   compatible with requested version "5.15.2".

I decided to got the /usr/lib64/cmake/ and change the version.cmake files, each time replacing 5.15.0 with 5.15.2
```
sudo nano /lib64/cmake/Qt5/Qt5ConfigVersion.cmake
sudo nano /lib64/cmake/Qt5AccessibilitySupport/Qt5AccessibilitySuppo
sudo nano /lib64/cmake/Qt5Bootstrap/Qt5BootstrapConfigVersion.cmake
sudo nano /lib64/cmake/Qt5Concurrent/Qt5ConcurrentConfigVersion.cmak
sudo nano /lib64/cmake/Qt5Core/Qt5CoreConfigVersion.cmake
sudo nano /lib64/cmake/Qt5DBus/Qt5DBusConfigVersion.cmake
sudo nano /lib64/cmake/Qt5Multimedia/Qt5MultimediaConfigVersion.cmake
sudo nano /lib64/cmake/Qt5Network/Qt5NetworkConfigVersion.cmake
```
At this point I got annoyed, contemplating giving up and decided to stop there and not change every Qt5 thing, thinking either Qt5 ot Qt5Core could be enough. I then ran

```
systemctl start systemd-timesyncd.service
sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
sudo pacman -R calamares
sudo pacman -Sy kconfig
git clone https://github.com/JPyke3/calamares-mbp
cd calamares-mbp
makepkg -s

```

Surprinsingly and pleasantly, this time the process went futher but then stopped with:

> 
> [ 30%] Linking CXX shared library ../../libcalamaresui.so
> [ 30%] Built target calamaresui
> make: *** [Makefile:160: all] Error 2
> ==> ERROR: A failure occurred in build().
>     Aborting..
Sorry, I didn't took more from the terminal, I've been very frustrated with my lack of knowledge and not being able to make it work so I didn't want to bother. I could provide screenshots if needed.

**4. One last desperate attempt**
Completly lost in a sea of terminal outputs I barely understood, I decided to run
```
sudo pacman -S calamares-mbp
sudo calamares
```
And got:
> calamares: error while loading shared libraries: libboost_python38.so.1.72.0: cannot open shared object file: No such file or directory.

So I sorta gave up for now. This is beyond my knowledge. I don't know how, but I hope this helps. I checked in /lib/ and libboost_python39 exists but not the libboost_python38.

**5. The Last stand**
I lied, I actually tried something else, same as 3 but instead of trying to compile from git I used "sudo pacman -S calamares-mbp". I didn't work either.

**tl;dr**
tried a bunch of stuff, like running pacman -Sy kconfig, manually chaning Qt5 version to 5.15.2 on the makefiles, but nothing worked.

**Comment by JPyke3** [comment]
Wow! Didn't know that this issue is as widespread as it is, sadly with my work commitments it's difficult to get some time to sit out and actually figure out what it is going on.

I unfortunately cannot for the life of me get this issue to replicate, but I will have my MacBook sitting beside me doing some work on this whist I work today, If I can get it to replicate and fix it, I will compile a new ISO image and send that one out to you guys.

Seems like you guys are working with KDE so I will use that image.

**Comment by guest-08** [comment]
No problem. Thanks a lot for all your good work! 

Let me know if I can provide you with more info that could help. 

Tschüss

**Comment by JPyke3** [comment]
So I took a look, cannot for the life of me replicate these issues, considering I'm not at home (Still away from Christmas time *lockdowns really suck*). 

Doubt I can take a real hard look at this until next week maybe the week after (I have no USB-C dongles or anything here, makes it a tad difficult).

What I can try do though is provide as much info as I can on the issue, as this issue is clearly not isolated.

`kpmcore` is the KDE Partition Manager core, you guys all seem to be having issues with `partition@partition` not working so that makes sense that this depend is the one stuffing everything up.

I saw on this discord message: https://discord.com/channels/595304521857630254/752437038405648404/795321675381473331 that kpmcore seems to be either broken or not in the Manjaro Repos anymore, I notice that there seems to be a `kpmcore3` package, but that is only version 3.0.2, but `calamares` requires > 4.2.0. So looks like for the sake of getting this to work `kpmcore4` will need to be manually installed.

To do this you will need to do the following (This is from a completley fresh restart on a live ISO):
```
systemctl start systemd-timesyncd.service
sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
sudo pacman -R calamares
sudo pacman -Syu
wget https://github.com/KDE/kpmcore/archive/v4.2.0.tar.gz
tar -xvf v4.2.0.tar.gz
cd kpmcore-4.2.0
mkdir build
cd build
cmake ..
make install
```

I am not sure if all the dependencies are in the Manjaro images, if they aren't let me know if this failed and I can figure out what is missing.
(Source https://github.com/KDE/kpmcore/blob/master/INSTALL.md)

After that, you should be able to do the following:

```
sudo pacman -S calamares-mbp
sudo calamares
```

and it *should* work,

Like I said, I have only tested this on the Family ubuntu machine we have lying around here. That's the best I can do until I get home. So any and all error output would be hugely appreciated for getting this sorted.

**Comment by guest-08** [comment]
No problem, I'm working from home due to restrictions and it's only getting worse on monday.

I have two bad news, sorry. I tried yesterday night using the i3 iso and got the same problem. I also tried what you just suggested now and still got the same issue. Could it be that problem is me? Maybe the way I partioned the drive? I could try with an external drive and a clean MacOS install. 
Here below are the steps I did with the associated screenshots. Unfortunatly, due to work and grad school I will not be able to mess around with this a lot until next week, so I hope I provided you with enough info here. Thanks again for your work and good luck with the lockdown.
Tchuss. 

----
**1. ISO Image**
I download the ISO from here :  https://github.com/JPyke3/mbp-manjaro/releases
I used balenaEtcher to write in on a USB drive. When the process is finished and I get the error message saying MacOS can't read the image I clicked on "Eject".

**2. Reboot and install kpmcore**
On the terminal I did

```
systemctl start systemd-timesyncd.service
sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
sudo pacman -R calamares
sudo pacman -Syu
wget https://github.com/KDE/kpmcore/archive/v4.2.0.tar.gz
tar -xvf v4.2.0.tar.gz
cd kpmcore-4.2.0
mkdir build
cd build
cmake ..
```
This is where I got my first error message.

![cmake_missing](https://user-images.githubusercontent.com/64798030/104737434-870b7600-5744-11eb-9d44-5aca0f609527.png)
(I typed and ran cmake .. several times cause I'm a dumbass and I could have misspelled it)

**3. Troubleshooting cmake**
I ran
`sudo pacman -S cmake` 
to then do cmake .. in kpmcore-4.2.0/build again and got the following error

![cmake_error](https://user-images.githubusercontent.com/64798030/104737691-dfdb0e80-5744-11eb-96e1-b832e731ad36.png)

I did

`sudo pacman -S extra-cmake-modules` 
And then the whole thing again and was greated with this follow error.

![cmake](https://user-images.githubusercontent.com/64798030/104737925-292b5e00-5745-11eb-9678-55ec1225d602.png)

So, I did
`sudo pacman -S pkgconfig` and did the whole thing again, this time successfully. 

![camke_success](https://user-images.githubusercontent.com/64798030/104738029-4b24e080-5745-11eb-8949-c3c7c862f76d.png)

I continued the process but make install didn't suceed.

![makeinstall_error](https://user-images.githubusercontent.com/64798030/104738089-5d068380-5745-11eb-8c21-0c0db22e09ee.png)

I restarted but this time with 
`sudo make install`

**4. Calamares**
Now I'm finally able to do
```
sudo pacman -S calamares-mbp
sudo calamares
```

And was greated by a familiar error message
![final_error](https://user-images.githubusercontent.com/64798030/104738265-97702080-5745-11eb-88a0-a2034b396a01.png)

**Comment by jamlam** [comment]
This worked for me with a slight change to the instructions (the install image is missing some of the build tools as noted above). The steps I used to start the installer were
```
systemctl start systemd-timesyncd.service
sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
sudo pacman -R calamares
sudo pacman -Sy cmake extra-cmake-modules pkgconfig
wget https://github.com/KDE/kpmcore/archive/v4.2.0.tar.gz
tar -xvf v4.2.0.tar.gz
cd kpmcore-4.2.0
mkdir build
cd build
cmake ..
sudo make install
sudo pacman -Sy calamares-mbp
sudo calamares
```

**Comment by a2ndTiberianBat** [comment]
This worked for me as well, thanks for the help!

**Comment by JPyke3** [comment]
Sweet thanks, hoping to get an image out ASAP with a correctly built version of kpmcore

**Comment by edgeux** [comment]
i could only get this to work with a boot from a USB stick. booting from a usb hdd seemed to not work. so anyone following along at home ... start from the very beginning and follow the steps carefully. no jumping in halfway.  nice work everyone. rah

**Comment by dominalien** [comment]
Right now cmake .. will fail complaining about wrong glibc version.
This is solved by installing lib32-glibc, which also pulls in glibc itself and then installation will proceed.

**Comment by AdityaGarg8** [comment]
> **Mac Model**
> MBP 16,1 (2019)
> 
> **Describe the bug**
> Cannot install Manjaro due to Calamares Initialization Failure.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. Boot into USB after flashing with iso (I used the 5.7 pre-release version (KDE) under 'download a latest release' on the installation wiki).
> 2. Execute Step 7 on the installation wiki to sync time and install calamares-mbp.
>    -- I had to use -Sy instead of -S when installing calamares-mbp in order to install it with pacman
> 3. Open the installer and see Calamares Initialization Failed Screen (could not load modules partition@partition)
> 
> If steps 2 and 3 are reversed (opening installer and selecting keyboard, then installing calamares-mbp) it fails during the install (after doing the partitioning) and relaunching the installer gives the same error screen.
> 
> **Notes**
> I used bootcamp to make a windows partition, then shrunk it to open enough unallocated space for manjaro (128gb) before starting the installation process.
> I used the refind boot manager as I could not see the flash drive on the one that pops up after holding the options key while booting.
> 
> **Screenshots**
> ![calamares_error](https://user-images.githubusercontent.com/56736018/104131872-5a6fec80-5347-11eb-9951-aa8a27c33276.png)

I am facing the same issue on GNOME. Any tips?

**Comment by AdityaGarg8** [comment]
> This worked for me with a slight change to the instructions (the install image is missing some of the build tools as noted above). The steps I used to start the installer were
> 
> ```
> systemctl start systemd-timesyncd.service
> sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
> sudo pacman -R calamares
> sudo pacman -Sy cmake extra-cmake-modules pkgconfig
> wget https://github.com/KDE/kpmcore/archive/v4.2.0.tar.gz
> tar -xvf v4.2.0.tar.gz
> cd kpmcore-4.2.0
> mkdir build
> cd build
> cmake ..
> sudo make install
> sudo pacman -Sy calamares-mbp
> sudo calamares
> ```

I guess it's for KDE and not GNOME

**Comment by AdityaGarg8** [comment]
```
systemctl start systemd-timesyncd.service
sudo sed -i 's/https:\/\/jacobpyke.xyz/https:\/\/mbp-repo.jacobpyke.xyz/' /etc/pacman.conf
sudo pacman -R calamares
sudo pacman -Sy cmake extra-cmake-modules pkgconfig
sudo pacman -Sy lib32-glibc gcc
wget https://github.com/KDE/kpmcore/archive/v4.2.0.tar.gz
tar -xvf v4.2.0.tar.gz
cd kpmcore-4.2.0
mkdir build
cd build
cmake ..
sudo make install
sudo pacman -Sy calamares-mbp
sudo calamares
```

This worked for KDE as well as GNOME for me. It resolved the cmake errors.
